How-To-Setup-Internal-Package-Repository-Nexus.md

```diff
@@ -0,0 +1,443 @@
+     * `UI: Base UI Privileges` and `UI: Repository Browser` - this will allow the user to log into the Nexus Dashboard and retrieve their API Key. Once this is done you can remove these roles if you do not want the user to be able to log in;
+     * `Repo: All NuGet Repositories (Full Control)` - this access allows the user to push packages to the repository;
+   * Click **OK**;
+   * Click **Save**;
+
+To use PowerShell we will again use the Nexus REST API:
+
+```powershell
+$json = @{
+    data = @{
+      userId       = "push-user"
+      firstName    = "Joe"
+      lastName     = "Bloggs"
+      email        = "joebloggs@example.com"
+      status       = "active"
+      password     = "password"
+      roles        = @{
+        '0' = 'nx-apikey-access'
+        '1' = 'nuget-all-full'
+        '2' = 'ui-basic'
+        '3' = 'ui-repo-browser'
+      }
+    }
+}
+
+# this is the default username and password for Nexus. Amend if you have changed that.
+# See https://help.sonatype.com/repomanager3/installation/post-install-checklist
+$username = 'admin'
+$password = 'admin123'
+$auth = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(("{0}:{1}" -f $username, $password)))
+
+$params = @{
+    Method      = 'Post'
+    Uri         = 'http://localhost:8081/nexus/service/local/users'
+    ContentType = 'application/json'
+    Headers     = @{ Authorization = "Basic $auth" }
+    Body        = ($json | ConvertTo-Json)
+}
+
+# invoke the Nexus API
+$result = Invoke-RestMethod @params
+
+# View the result after invoking the Nexus API
+$result.data
+
+# The result of that command should be:
+#
+# resourceURI : http://localhost:8081/nexus/service/local/users/push-user
+# userId      : push-user
+# password    : password
+# firstName   : Joe
+# lastName    : Bloggs
+# status      : active
+# email       : joebloggs@example.com
+# roles       : {nx-apikey-access, nuget-all-full, ui-basic, ui-repo-browser}
+```
+
+### Obtain Repository API Key
+
+As we mentioned earlier, a user must authenticate to the repository in order to be able to push packages to it. To do that they need to retrieve their API Key from the repository. We can do that from the browser or by using PowerShell.
+
+Follow these steps if you wish to use the browser:
+
+1. **Log In** as the user who will be pushing packages (created [above](#create-user-to-push-packages-to-the-repository));
+1. Click **Repositories** in the left hand column and then click the repository that the user will push packages to;
+1. Click the **NuGet** tab and:
+  ![NuGet Repository Personal API Key in Nexus](images/nexus/nuget-repository-personal-api-key.png)
+   * Note the URL in `Package Source`. We will need this later to push packages to;
+   * Click the **View API Key** button and then note the key shown in the `Personal API Key` field which we will call `<API KEY>` going forward;

```

Comment:
Is this Nexus v2? In Nexus v3 the API key is on the user itself. Not sure if it is also here.

Reply:
We should likely write this documentation from the perspective of NXRM v3.

Reply:
It is specific to the user - you need to log in as the user to get the API key. See Step 1 on line 222.